### PR TITLE
[NEW] Direct message from RoomInfo

### LINKED ISSUE #1511: [FEATURE] Direct Message from Channel Room itself
- Your Rocket.Chat Experimental app version: 4.2.0.99999999
- Your Rocket.Chat server version: 2.4.0 open.rocket.chat
- Device (or Simulator) you're running with: Android P

Its a feaure request. It would be helpful to be able to direct message someone from the channel room itself by tapping on his/her avatar. Right now, tapping on the avatar takes you to the user info page. So to direct message him/her, we have to search for the user first and then message. Instead, it would be helpful if there is direct messge option in the user info page like in case of web app. Or have a modal open up when avatar is tapped, showing option to direct message or view user info.

### PR DESCRIPTION
**Pull request by Prateek93a**
@RocketChat/ReactNative

Closes #1511

 I have implemented the direct message message feature. Right now the I have used the Button component available in the containers. If any changes regarding design or placement of the button or anything else, are needed, let me know, I will add them.
ezgif com-crop (4)

### CODE REVIEW

File: app/views/RoomInfoView/index.js [hunk 1]
```diff
@@ -259,6 +272,21 @@ class RoomInfoView extends React.Component {
 		return null;
 	}
 
+	renderMessageButton = () => {
+		const { roomUser } = this.state;
+		const { theme } = this.props;
+		return (
+			<Button
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: I think it's better if you use some Component like `ListItem`.
Captura de Tela 2020-01-13 às 10 24 20

File: app/views/RoomInfoView/index.js [hunk 2]
```diff
@@ -259,6 +272,21 @@ class RoomInfoView extends React.Component {
...
+				title={I18n.t('Message')}
+				type='primary'
+				onPress={() => this.goRoom({ name: roomUser.username })}
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Use only `this.goRoom` with the changes requested.

File: app/views/RoomInfoView/index.js [hunk 3]
```diff
@@ -141,6 +144,16 @@ class RoomInfoView extends React.Component {
 		}
 	}
 
+	goRoom = async({ name }) => {
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: `goRoom` not need any param.
You can use:
```
const { roomUser } = this.state;
const { username: name } = roomUser;
```

File: app/views/RoomInfoView/index.js [hunk 4]
```diff
@@ -259,6 +272,21 @@ class RoomInfoView extends React.Component {
 		return null;
 	}
 
+	renderMessageButton = () => {
+		const { roomUser } = this.state;
+		const { theme } = this.props;
+		return (
+			<Button
+				title={I18n.t('Message')}
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Use `Send_message` instead `Message`.

File: app/i18n/locales/en.js [hunk 5]
```diff
@@ -237,6 +237,7 @@ export default {
 	message: 'message',
 	messages: 'messages',
 	Messages: 'Messages',
+	Message: 'Message',
```
Comment (djorkaeffalexandre) [inline app/i18n/locales/en.js]: You can remove it.

**Comment by Prateek93a** [comment]
Ok Sir, On it.

**Comment by Prateek93a** [comment]
@djorkaeffalexandre Thanks for the review Sir. Now the button looks much better . Also Sir, when we go to room info screen from direct messaging, we still see the button. Should I add commit to hide button when user enters room info from direct message, or keep it like this only?
Screenshot_20200114-065500_ DEVELOP  RocketChatRN

File: app/views/RoomInfoView/index.js (hunk 1 above)
Comment (Prateek93a) [reply app/views/RoomInfoView/index.js]: Thanks for reviewing my code sir. I made the changes.

File: app/views/RoomInfoView/index.js [hunk 6]
```diff
@@ -1,10 +1,14 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { View, Text, ScrollView } from 'react-native';
+import {
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Unnecessary changes.

File: app/views/RoomInfoView/index.js [hunk 7]
```diff
@@ -259,6 +275,31 @@ class RoomInfoView extends React.Component {
 		return null;
 	}
 
+	renderDisclosure = () => {
+		const { theme } = this.props;
+		return <DisclosureImage theme={theme} />;
+	}
+
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Only 1 blank line, please.

File: app/views/RoomInfoView/index.js [hunk 8]
```diff
@@ -259,6 +275,31 @@ class RoomInfoView extends React.Component {
...
+
+	renderMessageButton = () => {
+		const { theme } = this.props;
+		return (
+			<>
+				<Separator theme={theme} />
+
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Not need this blank line.

File: app/views/RoomInfoView/index.js [hunk 9]
```diff
@@ -259,6 +275,31 @@ class RoomInfoView extends React.Component {
...
+				<ListItem
+					title={I18n.t('Send_message')}
+					onPress={this.goRoom}
+					showActionIndicator
+					right={this.renderDisclosure}
+					theme={theme}
+				/>
+
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Not need this blank line.

**Comment by Prateek93a** [comment]
Ok Sir, on it.

File: app/views/RoomInfoView/index.js (hunk 6 above)
Comment (Prateek93a) [reply app/views/RoomInfoView/index.js]: Sorry, I didn't remove the temprory changes. Now I have fixed them

**Comment by Prateek93a** [comment]
@djorkaeffalexandre Thanks for the review Sir, I have made the changes. Sorry , I didn't remove the temperory changes, I will surely review my code before pushing next time. Thanks

**Comment by Prateek93a** [comment]
@djorkaeffalexandre  Ok, so I should make changes with the margin, or should I change the appearance to make it like the native app version?

**Comment by djorkaeffalexandre** [comment]
@Prateek93a change appearance to make it like native app. :)

**Comment by Prateek93a** [comment]
Ok Sir, On it

**Comment by Prateek93a** [comment]
@djorkaeffalexandre  Is this ok?
Screenshot_20200212-231336_ DEVELOP  RocketChatRN

**Comment by djorkaeffalexandre** [comment]
Yes @Prateek93a ! Try to reduce the gap between the name and buttons.
It's getting good! Thanks.

**Comment by Prateek93a** [comment]
Ok, and also on the native app, username of the user is displayed, shall I add this also?
Screenshot_20200212-235838_ DEVELOP  RocketChatRN

**Comment by djorkaeffalexandre** [comment]
@Prateek93a, yes, it's good.

**Comment by Prateek93a** [comment]
Ok Sir, I will make the changes, clean the code and push the commit.

**Comment by Prateek93a** [comment]
Here I have pasted how room and user info view look in different themes. The gap between Timezone and buttons in dark theme in user info screen is because roles category is not visible. #1710 
Screenshot_20200213-114702_ DEVELOP  RocketChatRN
Screenshot_20200213-114832_ DEVELOP  RocketChatRN
Screenshot_20200213-114957_ DEVELOP  RocketChatRN
Screenshot_20200213-115057_ DEVELOP  RocketChatRN

**Comment by djorkaeffalexandre** [comment]
Ok @Prateek93a, awaiting your commits :)
Please, you can fix the roles color on this PR.

File: app/views/RoomInfoView/index.js [hunk 10]
```diff
@@ -21,10 +23,14 @@ import { withTheme } from '../../theme';
...
 const PERMISSION_EDIT_ROOM = 'edit-room';
-
 const camelize = str => str.replace(/^(.)/, (match, chr) => chr.toUpperCase());
-const getRoomTitle = (room, type, name, theme) => (type === 'd'
-	? <Text testID='room-info-view-name' style={[styles.roomTitle, { color: themes[theme].titleText }]}>{name}</Text>
+const getRoomTitle = (room, type, name, username, theme) => (type === 'd'
+	? (
+		<>
+			<Text testID='room-info-view-name' style={[styles.roomTitle, { color: themes[theme].titleText }]}>{ name }</Text>
+			<Text testID='room-info-view-name' style={[styles.roomUsername, { color: themes[theme].auxiliaryText }]}>{ username && `@${ username }` }</Text>
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Change `testID` to `room-info-view-username` and make the condition before render the `<Text>`.
Like this:  `{ username ? <Text>{username}</Text> : null }`.

File: app/views/RoomInfoView/index.js [hunk 11]
```diff
@@ -142,6 +148,20 @@ class RoomInfoView extends React.Component {
...
+		const { navigation } = this.props;
+		const result = await RocketChat.createDirectMessage(name);
+		if (result.success) {
+			await navigation.navigate('RoomsListView');
+			const rid = result.room._id;
+			navigation.navigate('RoomView', { rid, name, t: 'd' });
+		}
+	}
+
+	makeVideoCall = () => RocketChat.callJitsi(this.rid)
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: change not necessary

File: app/views/RoomInfoView/index.js [hunk 12]
```diff
@@ -260,6 +280,34 @@ class RoomInfoView extends React.Component {
 		return null;
 	}
 
+	renderDisclosure = () => {
+		const { theme } = this.props;
+		return <DisclosureImage theme={theme} />;
+	}
+
+	renderButton = (onPressHandler, iconName, text) => {
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: instead `onPressHandler` use only `onPress`, it's fine.

File: app/views/RoomInfoView/index.js [hunk 13]
```diff
@@ -272,20 +320,18 @@ class RoomInfoView extends React.Component {
...
-				{this.renderTimezone()}
-				{this.renderCustomFields(roomUser._id)}
-			</>
-		);
-	}
+	renderDirect = () => (
+		<>
+			{this.renderRoles()}
+			{this.renderTimezone()}
+			{this.renderCustomFields()}
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: You removed the `roomUser._id` param, please, revert them.

**Comment by Prateek93a** [comment]
@djorkaeffalexandre  Ok Sir, I have pushed commits for the above issue. I will push the commit to fix Roles color also.

**Comment by djorkaeffalexandre** [comment]
I requested some changes.
Before push your new commit, try to update your branch 👍

**Comment by Prateek93a** [comment]
Ok Sir

File: app/views/RoomInfoView/index.js (hunk 11 above)
Comment (Prateek93a) [reply app/views/RoomInfoView/index.js]: @djorkaeffalexandre Here, I didn't understand , should I remove `makeVideoCall`?

File: app/views/RoomInfoView/index.js (hunk 10 above)
Comment (Prateek93a) [reply app/views/RoomInfoView/index.js]: Done
